utils/filter.c

```diff
@@ -747,7 +805,10 @@ static int parse_hide_action(char *action, struct uftrace_trigger *tr,
 static int parse_clear_action(char *action, struct uftrace_trigger *tr,
 			      struct uftrace_filter_setting *setting)
 {
-	tr->flags |= TRIGGER_FL_CLEAR;
+	/* 'clear' suffix for any option other than -T/--trigger */
+	if (tr->flags)
+		tr->clear_flags = tr->flags;
```

Comment:
I'm afraid it can have other flags given together with the `@clear` action.  If you want to set the `orig_flags` only, then I think you need to do it in `setup_trigger_action`